File: 116.cpp
struct Solution {
    void make(Node *node){
        if (node->left == nullptr){
            return;
        }
        node->left->next = node->right;
        if (node->next != nullptr){
            node->right->next = node->next->left;
        }
        this->make(node->left);
        this->make(node->right);
    }
    inline Node* connect(Node* root) {
        if (root == nullptr){
            return nullptr;
        }
        this->make(root);
        return root;
    }
};

File: sol/116/116.py
class Solution:
    def connect(self, root: 'Optional[Node]') -> 'Optional[Node]':
        if root is None:
            return None
        def make(node):
            if node.left is None:
                return
            node.left.next = node.right
            if node.next is not None:
                node.right.next = node.next.left
            make(node.left)
            make(node.right)
        make(root)
        return root
